test/built-ins/Array/fromAsync/asyncitems-array-add-to-empty.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.fromasync
+description: >
+  Array.fromAsync respects array mutation
+includes: [asyncHelpers.js, compareArray.js]
+flags: [async]
+features: [Array.fromAsync]
+---*/
+
+asyncTest(async function () {
+  const items = [];
+  const promise = Array.fromAsync(items);
+  items.push(7);
+  const result = await promise;
+  assert.compareArray(result, []);
```

Comment:
Should this be `[7]` or is it intentionally omitted? I've been poring over the spec and don't see where an empty array is treated differently than an array with elements in it.

Reply:
Or is it something like, the first iterator `next()` happens before control returns to JS, while subsequent ones happen in following ticks? This and the other similar tests below could probably use a comment explaining what's going on.

Reply:
Thanks for the explanation, this helps a lot.